src/command_modules/azure-cli-ams/azure/cli/command_modules/ams/operations/account_filter.py

```diff
@@ -0,0 +1,111 @@
+                          timescale=None, bitrate=None, tracks=None, force_end_timestamp=None):
+
+    if not instance:
+        raise CLIError('The account filter resource was not found.')
+
+    if bitrate:
+        instance.first_quality = FirstQuality(bitrate=bitrate)
+
+    if any([start_timestamp, end_timestamp, presentation_window_duration,
+            live_backoff_duration, timescale, force_end_timestamp is not None]):
+
+        if instance.presentation_time_range is None:
+            if not all([start_timestamp, end_timestamp, presentation_window_duration,
+                        live_backoff_duration, timescale, force_end_timestamp is not None]):
+                raise CLIError('All parameters related to PresentationTimeRange must be set to create it.')
+
+            instance.presentation_time_range = PresentationTimeRange(
+                start_timestamp=start_timestamp,
+                end_timestamp=end_timestamp,
+                presentation_window_duration=presentation_window_duration,
+                live_backoff_duration=live_backoff_duration,
+                timescale=timescale,
+                force_end_timestamp=force_end_timestamp
+            )
+
+        else:
+            if start_timestamp is not None:
+                instance.presentation_time_range.start_timestamp = start_timestamp
+
+            if end_timestamp is not None:
+                instance.presentation_time_range.end_timestamp = end_timestamp
+
+            if presentation_window_duration is not None:
+                instance.presentation_time_range.presentation_window_duration = presentation_window_duration
+
+            if live_backoff_duration is not None:
+                instance.presentation_time_range.live_backoff_duration = live_backoff_duration
+
+            if force_end_timestamp is not None:
+                instance.presentation_time_range.force_end_timestamp = force_end_timestamp
+
+            if timescale is not None:
+                instance.presentation_time_range.timescale = timescale
+
+    if tracks is not None:
+        instance.tracks = _parse_filter_tracks_json(tracks)
+
+    return instance
+
+
+def _parse_filter_tracks_json(tracks):
+    tracks_result = None
+    if tracks is not None:
+        tracks_result = []
+        try:
+            tracks_json = json.loads(tracks)
+            for track_selection_json in tracks_json:
+                track_properties = []
+                for track_property_json in track_selection_json.get('trackSelections'):
+                    track_property = FilterTrackPropertyCondition(**track_property_json)
+                    track_properties.append(track_property)
+                tracks_result.append(FilterTrackSelection(track_selections=track_properties))
+        except TypeError as ex:
+            errorMessage = 'Malformed JSON.'
+            raise CLIError('{}. {}'.format(str(ex), errorMessage))
```

Comment:
I'm curious what the definition of "Malformed JSON" is in this case but it seems like maybe `FilterTrackPropertyCondition` and/or `FilterTrackSelection` are the ones that would raise a `TypeError`? The reason I ask is that `json.loads` usually throws either a `ValueError` in Python 2 or `JSONDecodeError` (subclass of `ValueError`) in Python 3.

* https://docs.python.org/2/library/json.html#json.loads
* https://docs.python.org/3/library/json.html#json.loads
```python
>>> import json
>>> json.loads('{')
Traceback (most recent call last):
  File "<input>", line 1, in <module>
  File "/home/.../.pyenv/versions/3.7.1/lib/python3.7/json/__init__.py", line 348, in loads
    return _default_decoder.decode(s)
  File "/home/.../.pyenv/versions/3.7.1/lib/python3.7/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/home/.../.pyenv/versions/3.7.1/lib/python3.7/json/decoder.py", line 353, in raw_decode
    obj, end = self.scan_once(s, idx)
json.decoder.JSONDecodeError: Expecting property name enclosed in double quotes: line 1 column 2 (char 1)

```
Of course a `TypeError` would be raised if `tracks` was None but the `if` statement above precludes that possibility?

```python
>>> json.loads(None)
Traceback (most recent call last):
  File "<input>", line 1, in <module>
  File "/home/.../.pyenv/versions/3.7.1/lib/python3.7/json/__init__.py", line 341, in loads
    raise TypeError(f'the JSON object must be str, bytes or bytearray, '
TypeError: the JSON object must be str, bytes or bytearray, not NoneType
```